snuba/query/parser/expressions.py

```diff
@@ -61,44 +73,94 @@ def visit_function_name(self, node: Node, visited_children: Iterable[Any]) -> st
     def visit_column_name(self, node: Node, visited_children: Iterable[Any]) -> Column:
         return Column(None, None, node.text)
 
-    def visit_empty(
+    def visit_low_pri_op(
+        self, node: Node, visited_children: Iterable[Any]
+    ) -> LowPriOperator:
+        if node.text == "+":
+            return LowPriOperator.PLUS
+        else:
+            return LowPriOperator.MINUS
+
+    def visit_high_pri_op(
         self, node: Node, visited_children: Iterable[Any]
-    ) -> Optional[Expression]:
-        return None
+    ) -> HighPriOperator:
+        if node.text == "*":
+            return HighPriOperator.MULTIPLY
+        else:
+            return HighPriOperator.DIVIDE
+
+    def visit_arithmetic_term(
+        self, node: Node, visited_children: Tuple[Any, Expression, Any]
+    ) -> Expression:
+        _, term, _ = visited_children
+
+        return term
 
     def visit_low_pri_arithmetic(
         self,
         node: Node,
-        visited_children: Tuple[Any, Expression, Any, Tuple[str, Expression]],
+        visited_children: Tuple[
+            Any, Expression, Any, Union[Node, Tuple[LowPriOperator, Expression]]
+        ],
     ) -> Expression:
         _, term, _, exp = visited_children
 
-        if exp is None:
-            # A check for any None child nodes
-            # that arose from empty space ''
+        if exp == "":
             return term
-        else:
-            # return exp[1] because
-            # exp[0] is a string which
-            # represents the operator
-            return plus(term, exp[1])
+
+        for elem in exp:
```

Comment:
I think the types you declared in input are still wrong.
1) `if exp == ""`. I assume this is the case where exp is a Node. You should not be able to compare it with a string. What does mypy say ?
2) `for elem in exp`: if exp is a Tuple it means you are iterating over the two elements of the tuple. So if this works, then exp is probably not a Tuple.